crates/soa/src/lib.rs

```diff
@@ -1,5 +1,9 @@
+#![cfg_attr(not(any(debug_assertions, test)), no_std)]
```

Comment:
Why no std? Are you going to pass an allocator in explicitly?

Reply:
Yeah, for new things I'm trying to use no_std as much as possible because for the caching strategy to work properly, it's crucial that all allocations happen in the arena so they can be serialized to/from disk trivially. no_std is an effective way to make sure the various passes aren't accidentally using the global allocator.